rosserial_python/src/rosserial_python/SerialClient.py

```diff
@@ -740,6 +740,13 @@ def send(self, topic, msg):
         Queues data to be written to the serial port.
         """
         self.write_queue.put((topic, msg))
+        """
+        Here need return a vaule .zzz
+        ServiceServer
+            callback
+                if self.parent.send(self.id, data_buffer.getvalue()) >= 0:
+        """
+        return 1
```

Comment:
The queue is not limited in length, so there is no way for this to fail. IMO the issue should be fixed in ServiceServer.